webapp/src/components/content/contentRegistry.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+/* eslint-disable react/require-optimization */
+import React from 'react'
+
+import {IntlShape} from 'react-intl'
+
+import {BlockTypes} from '../../blocks/block'
+import {IContentBlock, MutableContentBlock} from '../../blocks/contentBlock'
+import {MutableDividerBlock} from '../../blocks/dividerBlock'
+import {MutableImageBlock} from '../../blocks/imageBlock'
+import {MutableTextBlock} from '../../blocks/textBlock'
+import {Utils} from '../../utils'
+import DividerIcon from '../../widgets/icons/divider'
+import ImageIcon from '../../widgets/icons/image'
+import TextIcon from '../../widgets/icons/text'
+
+import DividerElement from './dividerElement'
+import ImageElement from './imageElement'
+import TextElement from './textElement'
+
+type RegistryEntry = {
+    type: BlockTypes,
+    createBlock: () => MutableContentBlock,
+    getDisplayText: (intl: IntlShape) => string,
+    getIcon: () => JSX.Element,
+    createComponent: (block: IContentBlock, readonly: boolean) => JSX.Element,
+}
+
+class ContentRegistry {
+    private registry: Map<BlockTypes, RegistryEntry> = new Map()
+
+    get contentTypes(): BlockTypes[] {
+        return [...this.registry.keys()]
+    }
+
+    constructor() {
+        this.registerContentType(
```

Comment:
I would expect the register calls being from outside. Something like `ContentRegistry` is a static class or there is a global `ContentRegistry` instance, and the ContentTypes defines themself and register themself at the end of the file.

Reply:
Moved registerContentType to the individual elements.